### DISCUSSION TITLE
Logical "OR" for required_scope

### DISCUSSION THREAD

**Discussion by cezarywysocki** [ory/oathkeeper#1038] [Ideas]
For our purpose we added feature, that changes how "required_scope" works.
Instead of logical "AND" (client_id must have all scopes listed) it turns to "OR" (client_id must have one, or more, of listed).
 Will you be interested to add this feature to official repository? Should we prepare PR?

Someone was looking for that already:
https://community.ory.sh/t/or-required-scope-with-oauth2-introspection/1533

**Comment by vinckr** [comment]
That sounds like a useful addition! Would you be open to write a short [design doc](https://github.com/ory/oathkeeper/issues/new?assignees=&labels=rfc&template=DESIGN-DOC.yml) for it?

**Reply by damianpietruszewski** [reply]
Design doc provided: https://github.com/ory/oathkeeper/issues/1129. Sorry for the delay
